include/supercluster.hpp

```diff
@@ -253,7 +263,16 @@ class Supercluster {
                 auto num_points = p.num_points;
                 point<double> weight = p.pos * double(num_points);
 
-                std::uint32_t id = (i << 5) + (zoom + 1);
+                // point_index encoded in the upper 27 bits
+                bool valid_point_index = ((i & 0x7ffffff) == i);
+#ifdef NDEBUG
+                if (!valid_point_index)
+                    return;
+#else
+                assert(valid_point_index);
+#endif
```

Comment:
You don't have to assert on every iteration, but rather assert `previous.clusters.size()` which `i` never exceeds.

Reply:
Good catch.